src/GraphQL/Types/Composite/ComplexGraphType.cs

```diff
@@ -418,9 +406,34 @@ public virtual FieldType AddField(FieldType fieldType)
         /// Adds a new field to the complex graph type and returns a builder for this newly added field.
         /// </summary>
         /// <typeparam name="TGraphType">The .NET type of the graph type of this field.</typeparam>
-        public virtual FieldBuilder<TSourceType, object> Field<TGraphType>()
+        /// <param name="name">The name of the field.</param>
+        public virtual FieldBuilder<TSourceType, object> Field<TGraphType>(string name = "default")
             where TGraphType : IGraphType
-            => Field<TGraphType, object>();
+            => Field<TGraphType, object>(name);
+
+        /// <summary>
+        /// Adds a new field to the complex graph type and returns a builder for this newly added field.
+        /// </summary>
+        /// <param name="type">The .NET type of the graph type of this field.</param>
+        /// <param name="name">The name of the field.</param>
+        public virtual FieldBuilder<TSourceType, object> Field(Type type, string name = "default")
+        {
+            var builder = CreateBuilder<object>(type).Name(name);
+            AddField(builder.FieldType);
+            return builder;
+        }
+
+        /// <summary>
+        /// Adds a new field to the complex graph type and returns a builder for this newly added field.
+        /// </summary>
+        /// <param name="type">The graph type of this field.</param>
+        /// <param name="name">The name of the field.</param>
+        public virtual FieldBuilder<TSourceType, object> Field(string name, IGraphType type)
```

Comment:
```suggestion
        public virtual FieldBuilder<TSourceType, object> Field(IGraphType type, string name)
```

I suggest we always have the type first -- so we have
- `Field<BooleanGraphType>("hello")`
- `Field(typeof(BooleanGraphType), "hello")`
- `Field(new BooleanGraphType(), "hello")`

Reply:
)))) I chose the best option for a long time. Now I would like to prefer name argument going first. It looks more naturally fo me to see field name first and then its type:

```graphql
type User {
  name: String
}
```

Reply:
Of course in case of generic argument type goes first because of such C# syntax.

Reply:
Didn't we just change this so the type is always first?

Reply:
I think the type should always be first for consistency.

Reply:
I guess the other logic makes sense also.....